src/google/protobuf/type.proto

```diff
@@ -65,125 +62,99 @@ message Field {
   // Kind represents a basic field type.
   enum Kind {
     // Field type unknown.
-    TYPE_UNKNOWN = 0;
-
+    TYPE_UNKNOWN        = 0;
     // Field type double.
-    TYPE_DOUBLE = 1;
-
+    TYPE_DOUBLE         = 1;
     // Field type float.
-    TYPE_FLOAT = 2;
-
+    TYPE_FLOAT          = 2;
     // Field type int64.
-    TYPE_INT64 = 3;
-
+    TYPE_INT64          = 3;
     // Field type uint64.
-    TYPE_UINT64 = 4;
-
+    TYPE_UINT64         = 4;
     // Field type int32.
-    TYPE_INT32 = 5;
-
+    TYPE_INT32          = 5;
     // Field type fixed64.
-    TYPE_FIXED64 = 6;
-
+    TYPE_FIXED64        = 6;
     // Field type fixed32.
-    TYPE_FIXED32 = 7;
-
+    TYPE_FIXED32        = 7;
     // Field type bool.
-    TYPE_BOOL = 8;
-
+    TYPE_BOOL           = 8;
     // Field type string.
-    TYPE_STRING = 9;
-
+    TYPE_STRING         = 9;
+    // Field type group (deprecated proto2 type)
+    TYPE_GROUP          = 10;
     // Field type message.
-    TYPE_MESSAGE = 11;
-
+    TYPE_MESSAGE        = 11;
     // Field type bytes.
-    TYPE_BYTES = 12;
-
+    TYPE_BYTES          = 12;
     // Field type uint32.
-    TYPE_UINT32 = 13;
-
+    TYPE_UINT32         = 13;
     // Field type enum.
-    TYPE_ENUM = 14;
-
+    TYPE_ENUM           = 14;
     // Field type sfixed32.
-    TYPE_SFIXED32 = 15;
-
+    TYPE_SFIXED32       = 15;
     // Field type sfixed64.
-    TYPE_SFIXED64 = 16;
-
+    TYPE_SFIXED64       = 16;
     // Field type sint32.
-    TYPE_SINT32 = 17;
-
+    TYPE_SINT32         = 17;
     // Field type sint64.
-    TYPE_SINT64 = 18;
-  }
+    TYPE_SINT64         = 18;
+  };
 
   // Cardinality represents whether a field is optional, required, or
   // repeated.
   enum Cardinality {
     // The field cardinality is unknown. Typically an error condition.
     CARDINALITY_UNKNOWN = 0;
-
     // For optional fields.
     CARDINALITY_OPTIONAL = 1;
-
     // For required fields. Not used for proto3.
     CARDINALITY_REQUIRED = 2;
-
     // For repeated fields.
     CARDINALITY_REPEATED = 3;
-  }
+  };
 
   // The field kind.
   Kind kind = 1;
-
   // The field cardinality, i.e. optional/required/repeated.
   Cardinality cardinality = 2;
-
   // The proto field number.
   int32 number = 3;
-
   // The field name.
   string name = 4;
-
   // The type URL (without the scheme) when the type is MESSAGE or ENUM,
   // such as `type.googleapis.com/google.protobuf.Empty`.
   string type_url = 6;
-
   // Index in Type.oneofs. Starts at 1. Zero means no oneof mapping.
   int32 oneof_index = 7;
-
   // Whether to use alternative packed wire representation.
   bool packed = 8;
-
   // The proto options.
   repeated Option options = 9;
+  // The JSON name for this field.
+  string json_name = 10;
```

Comment:
Can we assume this will always be populated (for any protos compiled after the change went in)? Basically it's not clear whether a runtime should be able to rely on this instead of having its own "name to JSON" conversion - I expect that's the intention, but it would be good to clarify.